addons/web/models/ir_http.py

```diff
@@ -95,18 +124,20 @@ def session_info(self):
 
     @api.model
     def get_frontend_session_info(self):
+        user = self.env.user
+        session_uid = request.session.uid
         session_info = {
-            'is_admin': request.session.uid and self.env.user._is_admin() or False,
-            'is_system': request.session.uid and self.env.user._is_system() or False,
-            'is_website_user': request.session.uid and self.env.user._is_public() or False,
-            'user_id': request.session.uid and self.env.user.id or False,
+            'is_admin': user._is_admin() if session_uid else False,
```

Comment:
It is not a war, `condition and true_value or false_value` was the only way before py2.5. It was confusing enough to change the language and introduce a real ternary operator. Since py2.5 (outside of consistency considerations) we should use `true_value of condition else false_value`.